wallet/src/Pos/Wallet/Web/Methods/History.hs

```diff
@@ -65,43 +66,49 @@ walletHistorySize =
     WalletHistorySize . fromIntegral . Map.size . unWalletHistory
 
 getFullWalletHistory
-    :: MonadWalletHistory ctx m
-    => CId Wal -> m (WalletHistory, WalletHistorySize)
-getFullWalletHistory cWalId = do
+    :: (MonadWalletHistory ctx m)
+    => WalletDB
+    -> CId Wal
+    -> m (WalletHistory, WalletHistorySize)
+getFullWalletHistory db cWalId = do
```

Comment:
Minor observation: we probably could have been even stricter and not pass `WalletDB` at all, but directly the `WalletSnapshot`, as I think we are not re-reading it twice.

Reply:
Ah, I have just noticed that in the call to `getHistory` we are asking the `WalletDB` directly, as if `MonadWalletHistory` had already an implicit `WalletDbReader`defined. Are we sure we want that?